### PR TITLE
Feat/distinguish decimal between native token and assets

### PR DESCRIPTION
**Pull request by rjman-ljm**
## Issues

Close
- #897

### CODE REVIEW

File: primitives/src/lib.rs [hunk 1]
```diff
@@ -101,7 +101,17 @@ pub trait PriceFeeder {
     fn get_price(asset_id: &CurrencyId) -> Option<PriceDetail>;
 }
 
-pub trait DecimalProvider {
+pub enum CurrencyType {
```
Comment (yz89) [inline primitives/src/lib.rs:104]: There is no need to add this type.

File: runtime/heiko/src/lib.rs [hunk 2]
```diff
@@ -973,14 +975,32 @@ impl DataProviderExtended<CurrencyId, TimeStampedPrice> for AggregatedDataProvid
     }
 }
 
-pub struct Decimal;
-impl DecimalProvider for Decimal {
+pub struct Currencies;
+impl CurrencyProvider<CurrencyId, CurrencyType> for Currencies {
+    fn get_currency_type(asset_id: &CurrencyId) -> CurrencyType {
```
Comment (yz89) [inline runtime/heiko/src/lib.rs:980]: We just need to distinguish whether the token is a native token or from pallet_asset.

File: pallets/loans/src/mock.rs [hunk 3]
```diff
@@ -173,7 +177,7 @@ impl pallet_prices::Config for Test {
     type FeederOrigin = EnsureRoot<AccountId>;
     type LiquidStakingExchangeRateProvider = LiquidStaking;
     type LiquidStakingCurrenciesProvider = LiquidStaking;
-    type Decimal = Decimal;
+    type CurrencyProvider = Currencies;
```
Comment (yz89) [inline pallets/loans/src/mock.rs:180]: No need to modify the Config.

File: runtime/heiko/src/lib.rs (hunk 2 above)
Comment (rjman-ljm) [reply runtime/heiko/src/lib.rs:980]: okay, got it.

**Review by GopherJ — APPROVED** [review]

**Review by yz89 — APPROVED** [review]

**Pull request merged**